### ISSUE TITLE
loss is nan

### ISSUE THREAD

**Issue by songh1024** [jwchoi384/Gaussian_YOLOv3#12]
Hi, nice work!

I use it to train yolov3-tiny.After training 150 batch, the loss became nan,Have you encountered this problem?  Thank you!
---------------------------------------------
batch=32,subdivisions=16,learning_rate=0.0001
----------------------------------------------
Region 15 Avg IOU: nan, Class: 0.441899, Obj: 0.010084, No Obj: 0.009767, .5R: 0.478261, .75R: 0.043478,  count: 23
Region 22 Avg IOU: 0.392249, Class: 0.575245, Obj: 0.002243, No Obj: 0.001650, .5R: 0.236842, .75R: 0.000000,  count: 38
Region 15 Avg IOU: nan, Class: 0.458486, Obj: 0.010441, No Obj: 0.009767, .5R: 0.428571, .75R: 0.000000,  count: 7
Region 22 Avg IOU: 0.464182, Class: 0.519555, Obj: 0.001945, No Obj: 0.001650, .5R: 0.460317, .75R: 0.015873,  count: 63
Region 15 Avg IOU: nan, Class: 0.461843, Obj: 0.010374, No Obj: 0.009767, .5R: 0.320000, .75R: 0.040000,  count: 25
Region 22 Avg IOU: 0.386944, Class: 0.502782, Obj: 0.001714, No Obj: 0.001650, .5R: 0.274510, .75R: 0.019608,  count: 51
Region 15 Avg IOU: nan, Class: 0.463781, Obj: 0.010159, No Obj: 0.009767, .5R: 0.285714, .75R: 0.071429,  count: 28
Region 22 Avg IOU: 0.439150, Class: 0.492993, Obj: 0.001829, No Obj: 0.001650, .5R: 0.428571, .75R: 0.000000,  count: 49
Region 15 Avg IOU: nan, Class: 0.466690, Obj: 0.010356, No Obj: 0.009767, .5R: 0.150000, .75R: 0.050000,  count: 20
Region 22 Avg IOU: 0.346722, Class: 0.426868, Obj: 0.001468, No Obj: 0.001650, .5R: 0.240000, .75R: 0.040000,  count: 50
Region 15 Avg IOU: nan, Class: 0.466256, Obj: 0.010191, No Obj: 0.009767, .5R: 0.136364, .75R: 0.000000,  count: 22
Region 22 Avg IOU: 0.382021, Class: 0.544101, Obj: 0.002096, No Obj: 0.001650, .5R: 0.247059, .75R: 0.000000,  count: 85
Region 15 Avg IOU: nan, Class: 0.465827, Obj: 0.010348, No Obj: 0.009767, .5R: 0.222222, .75R: 0.000000,  count: 18
Region 22 Avg IOU: 0.444984, Class: 0.587952, Obj: 0.001802, No Obj: 0.001650, .5R: 0.467532, .75R: 0.025974,  count: 77
Region 15 Avg IOU: nan, Class: 0.466738, Obj: 0.009839, No Obj: 0.009767, .5R: 0.300000, .75R: 0.000000,  count: 10
Region 22 Avg IOU: 0.371075, Class: 0.510235, Obj: 0.002036, No Obj: 0.001650, .5R: 0.300000, .75R: 0.000000,  count: 30
Region 15 Avg IOU: nan, Class: 0.460950, Obj: 0.010231, No Obj: 0.009767, .5R: 0.473684, .75R: 0.105263,  count: 19
Region 22 Avg IOU: 0.397543, Class: 0.496430, Obj: 0.001451, No Obj: 0.001650, .5R: 0.337662, .75R: 0.025974,  count: 77
Region 15 Avg IOU: nan, Class: 0.451978, Obj: 0.010417, No Obj: 0.009767, .5R: 0.666667, .75R: 0.166667,  count: 6
Region 22 Avg IOU: -nan, Class: -nan, Obj: -nan, No Obj: 0.001650, .5R: -nan, .75R: -nan,  count: 0
Region 15 Avg IOU: nan, Class: 0.463936, Obj: 0.010312, No Obj: 0.009767, .5R: 0.250000, .75R: 0.000000,  count: 28
Region 22 Avg IOU: 0.527799, Class: 0.639013, Obj: 0.001620, No Obj: 0.001650, .5R: 0.596774, .75R: 0.064516,  count: 62
Region 15 Avg IOU: nan, Class: 0.466003, Obj: 0.010128, No Obj: 0.009767, .5R: 0.240000, .75R: 0.040000,  count: 25
Region 22 Avg IOU: 0.493470, Class: 0.549164, Obj: 0.001179, No Obj: 0.001650, .5R: 0.461538, .75R: 0.128205,  count: 39
Region 15 Avg IOU: nan, Class: 0.464300, Obj: 0.010413, No Obj: 0.009767, .5R: 0.102564, .75R: 0.000000,  count: 39
Region 22 Avg IOU: 0.478209, Class: 0.595393, Obj: 0.001984, No Obj: 0.001650, .5R: 0.486111, .75R: 0.027778,  count: 72
Region 15 Avg IOU: nan, Class: 0.457656, Obj: 0.010366, No Obj: 0.009767, .5R: 0.240000, .75R: 0.000000,  count: 25
Region 22 Avg IOU: 0.451578, Class: 0.537413, Obj: 0.002011, No Obj: 0.001650, .5R: 0.385714, .75R: 0.028571,  count: 70
Region 15 Avg IOU: nan, Class: 0.457986, Obj: 0.010434, No Obj: 0.009767, .5R: 0.357143, .75R: 0.000000,  count: 14
Region 22 Avg IOU: 0.343575, Class: 0.520320, Obj: 0.002263, No Obj: 0.001650, .5R: 0.233333, .75R: 0.000000,  count: 30
Region 15 Avg IOU: nan, Class: 0.461090, Obj: 0.010450, No Obj: 0.009767, .5R: 0.222222, .75R: 0.000000,  count: 9
Region 22 Avg IOU: 0.510054, Class: 0.517130, Obj: 0.001078, No Obj: 0.001650, .5R: 0.550000, .75R: 0.000000,  count: 20
**151: nan, -nan avg, 0.000100 rate, 2.439035 seconds, 4832 images**

**Comment by KelvinHuang666** [comment]
I meet the same question,can U help me solve it,thanks!

**Comment by jwchoi384** [comment]
@songh1024 @KelvinHuang666  
In that case, try again. 
It is different each time you run it because weight is initialized randomly.
How many GPU did you use? 
What's your dataset?
Did you use cfg of tiny-YOLO for training?
If it fail again, increase the burn-in value in the cfg file.

**Comment by KelvinHuang666** [comment]
I use 4 GPU one time
I use my own dataset like voc (used in AlexeyAB project)
I didn't  use cfg of tiny-YOLO for training
I change your  cfg file (Gaussian_yolov3_BDD.cfg) for my dataset traiing
I test AlexeyAB project in 60000 batch,with loss < 1
I am starting new training with loss = 800 in 6600 batch,loss falling slowly
Thanks U

**Comment by songh1024** [comment]
@jwchoi384 @KelvinHuang666 :
I use 4 GPU one time and my own dataset.
I increase the burn-in value in the cfg file abd reduce learning rate.
Now traing with loss = 400 in 1300 batch, avg nan not appear.
Still in training

Thank You!

**Comment by bujianyiwang** [comment]
> I use 4 GPU one time
> I use my own dataset like voc (used in AlexeyAB project)
> I didn't use cfg of tiny-YOLO for training
> I change your cfg file (Gaussian_yolov3_BDD.cfg) for my dataset traiing
> I test AlexeyAB project in 60000 batch,with loss < 1
> I am starting new training with loss = 800 in 6600 batch,loss falling slowly
> Thanks U

how do you get a model use to realtime detect camera's video from network frame?

> I use 4 GPU one time
> I use my own dataset like voc (used in AlexeyAB project)
> I didn't use cfg of tiny-YOLO for training
> I change your cfg file (Gaussian_yolov3_BDD.cfg) for my dataset traiing
> I test AlexeyAB project in 60000 batch,with loss < 1
> I am starting new training with loss = 800 in 6600 batch,loss falling slowly
> Thanks U
how do you get a model use to realtime detect camera's video from network frame?
